lib/index.js

```diff
@@ -1,24 +1,32 @@
-var
+const
   Strategy = require('./passport/strategy');
 
-module.exports = function () {
-  this.context = {};
-  this.isDummy = true;
-  this.scope = {};
-
-  this.init = function (customConfig, context, isDummy) {
-    var strategy;
-    const defaultConfig = {
-      'strategies': {}
-    };
-    const config = Object.assign(defaultConfig, customConfig);
+/**
+ * @class PluginPassportOAuth
+ */
+class PluginPassportOAuth {
+  /**
+   * @constructor
+   */
+  constructor () {
+    this.context = {};
+    this.scope = {};
+    this.config = null;
+  }
+
+  init (customConfig, context) {
+    let strategy;
+    const
```

Comment:
nitpicking: for improved readability, I would put the defaultConfig as a global const variable